components/basic-filters/src/main/java/org/datacleaner/beans/filter/NullCheckFilter.java

```diff
@@ -132,12 +133,12 @@ public Query optimizeQuery(Query q, NullCheckCategory category) {
                 }
                 q.where(column, OperatorType.DIFFERENT_FROM, null);
                 if (considerEmptyStringAsNull && col.getDataType() == String.class) {
-                    q.where(column, OperatorType.DIFFERENT_FROM, "");
+                    q.having(FunctionType.TO_BOOLEAN, column, OperatorType.EQUALS_TO, "TRUE");
```

Comment:
You are right. It does not work. Now it even gives me an exception when run against oracle db. I don't know what I tested yesterday that it worked.